docs/src/pages/components/data-grid/columns/columns.md

```diff
@@ -79,17 +85,185 @@ Alternatively, to disable all columns resize, set the prop `disableColumnResize=
 
 {{"demo": "pages/components/data-grid/columns/ColumnSizingGrid.js", "disableAd": true, "bg": "inline"}}
 
-<!--
-- https://ag-grid.com/javascript-grid-resizing/
-- https://demos.telerik.com/kendo-ui/grid/column-resizing
-- https://www.telerik.com/kendo-react-ui/components/grid/columns/resizing/
-- https://elastic.github.io/eui/#/tabular-content/data-grid
-- https://devexpress.github.io/devextreme-reactive/react/grid/docs/guides/column-resizing/
-- http://schrodinger.github.io/fixed-data-table-2/example-resize.html
-- https://ej2.syncfusion.com/react/demos/#/material/grid/column-resizing
-- https://js.devexpress.com/Demos/WidgetsGallery/Demo/DataGrid/ColumnResizing/React/Light/
-- https://www.jqwidgets.com/react/react-grid/#https://www.jqwidgets.com/react/react-grid/react-grid-columnsresize.htm
--->
+### Value getter
+
+Sometimes a column might not have a corresponding value, or you might want to render a combination of different fields.
+
+To achieve that, set the `valueGetter` attribute of `GridColDef` as in the example below
+
+**Note**: You need to set a `sortComparator` for the column sorting to work when setting the `valueGetter` attribute.
+
+```tsx
+function getFullName(params: ValueGetterParams) {
+  return `${params.getValue('firstName') || ''} ${
+    params.getValue('lastName') || ''
+  }`;
+}
+
+const columns: GridColDef[] = [
+  { field: 'firstName', headerName: 'First name', width: 130 },
+  { field: 'lastName', headerName: 'Last name', width: 130 },
+  {
+    field: 'fullName',
+    headerName: 'Full name',
+    width: 160,
+    valueGetter: getFullName,
+    sortComparator: (v1, v2, cellParams1, cellParams2) =>
+      getFullName(cellParams1).localeCompare(getFullName(cellParams2)),
+  },
+];
+```
+
+{{"demo": "pages/components/data-grid/columns/ValueGetterGrid.js", "bg": "inline"}}
+
+The value generated is used for filtering, sorting, rendering, etc unless overridden by a more specific configuration.
+
+### Value formatter
+
+The value formatters allow you to format values for display as a string.
+For instance, you can use it to format a JavaScript `Date` object.
+
+```tsx
+const columns: GridColDef[] = [
+  {
+    field: 'date',
+    headerName: 'Year',
+    valueFormatter: (params: ValueFormatterParams) =>
+      (params.value as Date).getFullYear(),
+  },
+];
+```
+
+{{"demo": "pages/components/data-grid/columns/ValueFormatterGrid.js", "bg": "inline"}}
+
+The value generated is used for filtering, sorting, rendering in the cell and outside, etc unless overridden by a more specific configuration.
+
+### Render cell
+
+By default, the grid render the value as a string in the cell.
+It resolves the rendered output in the following order:
+
+1. `renderCell() => ReactElement`
```

Comment:
this notation doesn't really make sense. Might as well put the proper type and tsx notation?

Reply:
True, sounds like a good topic for a follow-up